src/umpire/strategy/mixins/AlignedAllocation.inl

```diff
@@ -17,28 +17,30 @@ namespace mixins {
 
 inline std::size_t AlignedAllocation::aligned_round_up(std::size_t size)
 {
-    return size + (m_alignment - 1) - (size - 1) % m_alignment;
+  return size + (m_alignment - 1) - (size - 1) % m_alignment;
 }
 
 inline void* AlignedAllocation::aligned_allocate(std::size_t size)
 {
-    std::size_t total_bytes{ size + m_alignment - 1 };
-    uintptr_t ptr{ reinterpret_cast<uintptr_t>(m_allocator->allocate(total_bytes)) };
+  std::size_t total_bytes{ size + m_alignment };
+  uintptr_t ptr{ reinterpret_cast<uintptr_t>(m_allocator->allocate(total_bytes)) };
 
-    uintptr_t alignment{ m_alignment - 1 };
-    uintptr_t mask{ m_mask };
-    void *aligned_ptr{ reinterpret_cast<void*>( (ptr + alignment) & mask ) };
+  UMPIRE_POISON_MEMORY_REGION(m_allocator, reinterpret_cast<void*>(ptr), total_bytes);
```

Comment:
Isn't memory already marked for ASAN in top-level allocation strategies like QuickPool? You could say that you poison memory only when you're sure this is the low-level allocator (e.g. CudaMallocAllocator). Maybe I don't understand everything here.

Reply:
Yes, but some of this memory may have been missed if an alignment would have been necessary.  It does seem wasteful to poison memory twice however.  I think I will change this to continue to poison the memory region here in the mixin since it knows the base pointer and total size of what was allocated.  I will be providing an update to the pools to no longer perform the redundant poison after teh aligned_allocate call.

Reply:
> some of this memory may have been missed if an alignment would have been necessary

I'm not sure what you mean by that. I think we should poison the whole memory, not only the aligned part. Doing that in the lowest level (i.e. the one where cudaMalloc is actually called) guarantees that the whole memory is poisoned, and that poisoning is not called twice.

Reply:
You are correct.  However, for a pool strategy, this is the lowest level.  Let me try to explain:

If an Allocator is built directly on a resource (CudaMallocAllocator), then the memory allocated must not be poisoned because the application calling allocate will start to use the memory right away.

If an Allocator is built on top of a pool strategy, then the pool will return memory from the pool it is managing.  This pool will separately be filled by chunk allocations from CudaMallocAllocator when needed.  When new chunks are given to the pool, they will more thank likely be (much) larger than the calling application is requesting.  So, those chunks should initially be given to the pool as poisoned.

Reply:
So, the lowest level allocator will be called by two consumers.  By the application directly and by a pool.  When the lowest level allocator is called directly from the application, it cannot be poisoned.  When it is called by the pool, it should be poisoned.

Does this make sense?

Reply:
Makes sense! Thank you for that explanation 😊